src/CharacterPositionController.js

```diff
@@ -0,0 +1,234 @@
+// @flow
+
+import { injectIntl } from 'react-intl';
+import type {IntlShape} from 'react-intl';
+import React from 'react';
+import CharacterState from './CharacterState';
+import classNames from 'classnames';
+import { ReactComponent as RobotIcon } from './svg/Robot.svg';
+import { ReactComponent as SpaceShipIcon } from './svg/SpaceShip.svg';
+import { ReactComponent as RabbitIcon } from './svg/Rabbit.svg';
+import { ReactComponent as MovePositionUp } from './svg/MovePositionUp.svg';
+import { ReactComponent as MovePositionRight } from './svg/MovePositionRight.svg';
+import { ReactComponent as MovePositionDown } from './svg/MovePositionDown.svg';
+import { ReactComponent as MovePositionLeft } from './svg/MovePositionLeft.svg';
+import { ReactComponent as TurnPositionRight } from './svg/TurnPositionRight.svg';
+import { ReactComponent as TurnPositionLeft } from './svg/TurnPositionLeft.svg';
+import './CharacterPositionController.scss';
+
+type CharacterPositionControllerProps = {
+    intl: IntlShape,
+    characterState: CharacterState,
+    editingDisabled: boolean,
+    world: string,
+    onChangeCharacterPosition: (direction: ?string) => void,
+    onChangeCharacterXPosition: (columnLabel: string) => void,
+    onChangeCharacterYPosition: (rowLabel: string) => void
+};
+
+type CharacterPositionControllerState = {
+    prevPropsCharacterState: CharacterState,
+    characterColumnLabel: string,
+    characterRowLabel: string
+};
+
+class CharacterPositionController extends React.Component<CharacterPositionControllerProps, CharacterPositionControllerState> {
+    constructor(props: CharacterPositionControllerProps) {
+        super(props);
+        this.state = {
+            prevPropsCharacterState: this.props.characterState,
+            characterColumnLabel: this.props.characterState.getColumnLabel(),
+            characterRowLabel: this.props.characterState.getRowLabel()
+        }
+    }
+
+    static getDerivedStateFromProps(props: CharacterPositionControllerProps, state: CharacterPositionControllerState) {
+        if (props.characterState !== state.prevPropsCharacterState) {
+            const currentCharacterState = props.characterState;
+            return {
+                prevPropsCharacterState: currentCharacterState,
+                characterColumnLabel: currentCharacterState.getColumnLabel(),
+                characterRowLabel: currentCharacterState.getRowLabel()
+            };
+        } else {
+            return null;
+        }
+    }
+
+    handleClickCharacterPosition = (e) => {
```

Comment:
We can name the methods of this component more consistently and more clearly to better communicate their responsibility and relationship:

* `handleClickCharacterPosition` -> `handleClickCharacterPositionButton`
* `handleKeyDownCharacterPosition` -> `handleKeyDownCharacterPositionButton`
* `handleChangeCharacterPosition` -> I think we can just remove this method and call `this.props.onChangeCharacterPosition` directly in the two methods above
* `handleChangeCharacterPositionLabel` -> fine as is
* `handleBlurCharacterPositionBox` -> `handleBlurCharacterPositionLabel`
* `handleUpdateCharacterPosition` -> `handleKeyDownCharacterPositionLabel`